library/std/src/sys/windows/fs.rs

```diff
@@ -550,6 +556,14 @@ impl File {
         })?;
         Ok(())
     }
+
+    pub fn set_times(&self, times: FileTimes) -> io::Result<()> {
+        cvt(unsafe {
+            c::SetFileTime(self.as_handle(), None, Some(&times.accessed), Some(&times.modified))
```

Comment:
On Windows, [`SetFileTime`](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-setfiletime#parameters) silently doesn't set the timestamp if the `FILETIME` has the value zero (dwLowDateTime and dwHighDateTime members are 0). That's not likely to be a common situation but looking at the code in src/sys/windows/time.rs, I don't see anything that prevents it, so that's a situation where the current code here doesn't behave as documented.

Reply:
I'm making intentional use of that behavior here, to not set the times the user hasn't set.

The case you're concerned about is intentionally trying to set a time corresponding to zero? I can add documentation for that.

Reply:
Yes, I'm thinking of users trying to set a time corresponding to zero. For example,
```
     let time = SystemTime::UNIX_EPOCH.checked_sub(Duration::new(11644473600, 0)).unwrap();
```
produces a zero `SystemTime` on Windows. Passing that to `set_times` here silently doesn't set the time. `set_tiimes` returns an `io::Result`, so it would be feasible for it to report an error in this situation instead.